moped-editor/src/queries/project.js

```diff
@@ -192,19 +206,48 @@ export const TEAM_QUERY = gql`
   }
 `;
 
-export const UPSERT_PROJECT_PERSONNEL = gql`
-  mutation UpsertProjectPersonnel(
-    $objects: [moped_proj_personnel_insert_input!]!
+export const INSERT_PROJECT_PERSONNEL = gql`
+  mutation InserProjectPersonnel($object: moped_proj_personnel_insert_input!) {
+    insert_moped_proj_personnel_one(object: $object) {
+      project_personnel_id
+    }
+  }
+`;
+
+export const UPDATE_PROJECT_PERSONNEL = gql`
+  mutation UpdateProjectPersonnel(
+    $deleteIds: [Int!]
+    $id: Int!
+    $updatePersonnelObject: moped_proj_personnel_set_input!
+    $addRolesObjects: [moped_proj_personnel_roles_insert_input!]!
   ) {
-    insert_moped_proj_personnel(
-      objects: $objects
-      on_conflict: {
-        constraint: moped_proj_personnel_project_id_user_id_role_id_key
-        update_columns: [project_id, user_id, role_id, is_deleted, notes]
-      }
+    update_moped_proj_personnel_by_pk(
+      pk_columns: { project_personnel_id: $id }
+      _set: $updatePersonnelObject
+    ) {
+      project_personnel_id
+    }
+    update_moped_proj_personnel_roles(
+      where: { id: { _in: $deleteIds } }
+      _set: { is_deleted: true }
     ) {
       affected_rows
     }
+    insert_moped_proj_personnel_roles(objects: $addRolesObjects) {
+      affected_rows
+    }
+  }
+`;
+
+export const DELETE_PROJECT_PERSONNEL = gql`
+  mutation DeleteProjectPersonnel($id: Int!) {
+    update_moped_proj_personnel_by_pk(
+      pk_columns: { project_personnel_id: $id }
+      _set: { is_deleted: true }
```

Comment:
if we delete a team member (`moped_proj_personnel`), we do not delete their team member roles.

that seems like the right way to do this? if for some reason we wanted to revert a team member delete, its roles state would be intact.